skbio/io/format/blast6.py

```diff
@@ -0,0 +1,287 @@
+0  subject2  query1    100          NaN        0  subject2     32   100
+1  subject1  query2     70          NaN        0  subject1     19    70
+2  subject2  query1    100          NaN        0  subject2     18    50
+
+References
+----------
+.. [1] http://blastedbio.blogspot.com/2014/11/column-headers-in-blast-tabular-\
+and-csv.html
+.. [2] http://www.ncbi.nlm.nih.gov/books/NBK279675/
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from skbio.io import create_format
+import pandas as pd
+import numpy as np
+
+blast6 = create_format('blast+6')
+
+_possible_columns = {'qseqid': str, 'qgi': int, 'qacc': str, 'qaccver': str,
+                     'qlen': int, 'sseqid': str, 'sallseqid': str, 'sgi': int,
+                     'sallgi': int, 'sacc': str, 'saccver': str,
+                     'sallacc': str, 'slen': int, 'qstart': int, 'qend': int,
+                     'sstart': int, 'send': int, 'qseq': str, 'sseq': str,
+                     'evalue': float, 'bitscore': float, 'score': int,
+                     'length': int, 'pident': float, 'nident': int,
+                     'mismatch': int, 'positive': int, 'gapopen': int,
+                     'gaps': int, 'ppos': float, 'frames': str, 'qframe': int,
+                     'sframe': int, 'btop': int, 'staxids': str,
+                     'sscinames': str, 'scomnames': str, 'sblastnames': str,
+                     'sskingdoms': str, 'stitle': str, 'salltitles': str,
+                     'sstrand': str, 'qcovs': int, 'qcovhsp': int}
+_default_columns = ['qseqid', 'sseqid', 'pident', 'length', 'mismatch',
+                    'gapopen', 'qstart', 'qend', 'sstart', 'send',
+                    'evalue', 'bitscore']
+
+
+@blast6.reader(pd.DataFrame)
+def _blast6_to_data_frame(fh, columns=None, default_columns=False):
+    if default_columns is True and columns is not None:
+        raise ValueError("You cannot use the default columns and also create"
+                         " your own columns")
+    if default_columns is False and columns is None:
+        raise ValueError("You must specify whether default columns or custom"
+                         " columns are being used.")
+    if columns is not None:
+        names = columns
+    else:
+        names = _default_columns
+
+    df = pd.read_csv(fh, na_values='N/A', sep='\t', header=None)
+
+    if len(list(df.columns.values)) != len(names):
+        raise ValueError("The amount of data you have supplied does not match"
+                         " the number of columns.")
+
+    try:
```

Comment:
Can you move the try/except to wrap the line(s) that could throw this error? I'm guessing it's the `.astype()` line. This will make it clearer where exactly we're trying to catch the pandas error and avoids catching other `ValueErrors` that may be raised by other parts of the code.